### PR TITLE
Test against AGP 8.7.0-alpha06

### PR DESCRIPTION
**Pull request by MGaetan89**
This PR updates the 8.6.0 branch to rc01, and adds the 8.7.0 branch of AGP to the test matrix.

### CODE REVIEW

**Comment by MGaetan89** [comment]
It looks like there are failing tests with AGP 8.7.0. I'll have a closer look at it this weekend.

**Comment by autonomousapps** [comment]
> do you think it would make sense to extract the test matrix out of the project's code, into the GitHub Actions workflow file?

I used to do this, ages and ages ago. I have the current setup because I wanted the full matrix encapsulated in the build system so I could easily run all the tests locally without having to invoke a bash script. This was before there were so many tests I risked waiting till the heat death of the universe for the suite to complete (I never run the full suite locally anymore).

tl;dr I'm open to a change, but it'll be a higher barrier to merge.

File: src/functionalTest/groovy/com/autonomousapps/android/AbstractAndroidSpec.groovy [hunk 1]
```diff
@@ -41,6 +42,7 @@ abstract class AbstractAndroidSpec extends AbstractFunctionalSpec {
     AGP_8_4,
     AGP_8_5,
     AGP_8_6,
+    AGP_8_7,
```
Comment (autonomousapps) [inline src/functionalTest/groovy/com/autonomousapps/android/AbstractAndroidSpec.groovy:45]: The problem is this is too many versions. CI can't handle this load. We should test against the min version (8.0), the current stable version (8.5.1), and maybe also the latest alpha.

**Review by autonomousapps — CHANGES_REQUESTED** [review]

File: src/functionalTest/groovy/com/autonomousapps/android/AbstractAndroidSpec.groovy (hunk 1 above)
Comment (MGaetan89) [reply src/functionalTest/groovy/com/autonomousapps/android/AbstractAndroidSpec.groovy:40]: Thanks for the tip!
I've updated the PR to test only against AGP 8.0 (min supported stable version), 8.5 (max supported stable version), 8.7 (latest alpha version).

Comment (MGaetan89) [reply src/functionalTest/groovy/com/autonomousapps/android/AbstractAndroidSpec.groovy:40]: It's still failing. I think it was fine running the tests for every AGP versions, but there is probably some breaking change in AGP 8.7

Comment (MGaetan89) [reply src/functionalTest/groovy/com/autonomousapps/android/AbstractAndroidSpec.groovy:45]: It looks like AGP 8.7 requires Gradle 8.9. So I've updated the project to use Gradle 8.9 (previously 8.7) and made the CI run tests for every AGP version again.
Let me know if you still prefer to keep only the min stable version, max stable version and the latest alpha version.

Comment (MGaetan89) [reply src/functionalTest/groovy/com/autonomousapps/android/AbstractAndroidSpec.groovy:45]: It seems that everything is working fine now.
- Is it ok to upgrade to Gradle 8.9, or do you prefer a separate PR?
- Should I keep the tested AGP versions list as it is now? Or only the three versions you suggested?

Comment (autonomousapps) [reply src/functionalTest/groovy/com/autonomousapps/android/AbstractAndroidSpec.groovy:45]: I think updating to Gradle 8.9 would be better in a separate PR.

For this PR, you could just update the AGP/Gradle compatibility matrix, and then it won't try to run the test against an incompatible combination.

Comment (MGaetan89) [reply src/functionalTest/groovy/com/autonomousapps/android/AbstractAndroidSpec.groovy:45]: I've created the PR to update Gradle here: https://github.com/autonomousapps/dependency-analysis-gradle-plugin/pull/1236

I'll update this PR soon too.

Comment (MGaetan89) [reply src/functionalTest/groovy/com/autonomousapps/android/AbstractAndroidSpec.groovy:40]: It should be good now. Let me know if I missed something.

**Review by autonomousapps — APPROVED** [review]
LGTM
